src/main/java/ooo/oshi/foreign/windows/Kernel32Library.java

```diff
@@ -92,5 +94,130 @@ public static String getComputerName() {
+			throw new RuntimeException(e);
+		}
+	}
+
+	private static final MethodHandle closeHandle = methodHandle("CloseHandle",
+			FunctionDescriptor.of(JAVA_BOOLEAN, ADDRESS));
+
+
+	/**
+	 * Opens an existing local process object.
+	 *
+	 * @param desiredAccess The access to the process object.
+	 * @param inheritHandle Processes created by this process should inherit the handle
+	 * @param processId The identifier of the local process to be opened.
+	 * @return the open handle to the specified process.
+	 */
+	public static Addressable openProcess(int desiredAccess, boolean inheritHandle, int processId) {
+		try {
+			Addressable hProcess = (MemoryAddress) openProcess.invokeExact(desiredAccess, inheritHandle, processId);
+			if (hProcess == null) {
+				throw new Win32Exception(getLastError());
+			}
+			return hProcess;
+		} catch (Throwable e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	private static final MethodHandle openProcess = methodHandle("OpenProcess",
+			FunctionDescriptor.of(ADDRESS, JAVA_INT, JAVA_BOOLEAN, JAVA_INT));
+
+	private static final String queryFullProcessImageName(Addressable hProcess, int dwFlags) {
+		try {
+			SegmentAllocator allocator = SegmentAllocator.implicitAllocator();
+			int size = WinBase.MAX_PATH; // Start with MAX_PATH, then increment with 1024 each iteration
+			// TODO: Handle case if process image length exceeds predefined buffer size
+			Addressable lpExeName = allocator.allocate(size);
+			Addressable lpdwSize = allocator.allocate(JAVA_INT, size);
+			if (!(boolean) queryFullProcessImageName.invokeExact(hProcess, dwFlags, lpExeName, lpdwSize)) {
+				throw new Win32Exception(getLastError());
+			}
+			byte[] bytes = ((MemorySegment) lpExeName).toArray(ValueLayout.JAVA_BYTE);
+			return ParseUtil.parseByteArrayToUtf16(bytes);
+		} catch (Throwable e) {
+			throw new RuntimeException(e);
+		}
+    }
+
+	private static final MethodHandle queryFullProcessImageName = methodHandle("QueryFullProcessImageNameW",
+            FunctionDescriptor.of(JAVA_BOOLEAN, ADDRESS, JAVA_INT, ADDRESS, ADDRESS));
+
+	/**
+	 * Retrieves the full name of the executable image for the specified process.
+	 *
+	 * @param pid The identifier of the local process to be opened.
+	 * @param dwFlags flags for path format of returned path.
+	 * @return the full name of the executable image for the specified process.
+	 */
+	public static final String queryFullProcessImageName(int pid, int dwFlags) {
+		Addressable hProcess = null;
+		Win32Exception we = null;
+
+		try {
+			hProcess = (Addressable) openProcess(WinBase.PROCESS_QUERY_INFORMATION | WinBase.PROCESS_VM_READ, false,
+					pid);
+			if (hProcess == null) {
+				throw new Win32Exception(getLastError());
+			}
+			return queryFullProcessImageName(hProcess, dwFlags);
+		} catch (Win32Exception e) {
+			we = e;
+			throw we; // re-throw to invoke finally block
+		} finally {
+			try {
+				closeHandle(hProcess);
+			} catch (Win32Exception e) {
+				if (we == null) {
```

Comment:
@dbwiddis, I've implemented a similar error handling as the JNA library for this method. Seems like the above finding is not critical. Do you have any suggestions for this? Thanks!

Reply:
Sorry for slow review, crazy busy week.  A few thoughts:
 - the sonatype-lift bot annoys me sometimes.  I do wonder how a thrown exception can possibly be null, though! 
 - I'm not so sure about implementing the JNA named exceptions.  I (we as a community) haven't really even discussed custom exceptions yet.  This is probably ok for now but probably won't stick.
 - I'm not sure how many of these functions are used by OSHI.... it's totally fine to add them for educational value but the goal here is not to reproduce JNA it is to reproduce OSHI.  So ideally you can find Kernel32 functions that OSHI calls somewhere in its Windows code.

Reply:
No worries at all. In fact, I really appreciate your detailed review comments.

- So the `queryFullProcessImageName()` method involves calls to OpenProcess, QueryFullProcessImageName and CloseHandle win32 API. It is possible for any one or all of them to fail. So if only the CloseHandle API fails, `we` would be `null`. Say if the `OpenProcess()` or `QueryFullProcessImageName()` API fails and `CloseHandle()` succeeds, `we` would hold the win32 exception and eventually that would be thrown. Now, in case `OpenProcess()` or `QueryFullProcessImageName()` fails and `CloseHandle()` fails as well,  `we` would again hold the win32 exception of `OpenProcess()` or `QueryFullProcessImageName()` call (whichever failed) and the exception thrown by `CloseHandle() `would be suppressed.
- Got it! Totally agree with you. I implemented it to simplify things. I'd definitely be open to a better or more standardised way of handling exceptions.
- Understood! In fact, my original intention was to implement the WindowsOSProcess class, but it seemed a bit overwhelming to me being a first-time contributor to this project. So I started with some of the JNA-based util methods.  So far I see that GetComputerName, OpenProcess, QueryFullProcessImageName and CloseHandle are being used by OSHI. GetTempPathW probably isn't used and I'd defer to you on whether it should be kept or removed.

Thanks again for the helpful feedback!

Reply:
I probably won’t have time for a full review till the weekend.  Thanks for all the work you are doing.  

The PDH functions (from Psapi) are a really high priority if you are looking for more. :)

Reply:
Sure! Thanks for the tip. I'll dig into that 😄

Reply:
I've got a bit more time to look into this but I seem to have lost track of where we are with this PR. Is it ready for review or are you still working on things?  Quick glance at the file diff shows me the "Win32Exception" I'd rather not be permanent but I'm ok merging if we plan on removing it later.

Let me know what you'd like to do with this PR (push it through and modify later, or iterate on it.)

Reply:
I agree with you about the Win32Expection. To prevent confusion and backlog of removing it later, I think I'll get eliminate the Win32Expection and thrown a basic Runtime Exception for now. Hope that's okay? If yes, I'll make the changes and push them in a couple of days (Sorry, I'm a bit caught up with work and personal commitments 😞 )

Reply:
I've pushed the changes to replace the custom exception class with a standard Exception.